src/ShopBundle/objects/TCMSSmartURLHandler/TPkgShopRouteControllerArticle.class.php

```diff
@@ -331,53 +331,54 @@ private function processCategoryResponse($aResponse, $request, $cacheKey = null)
      */
     public function shopCategory(Request $request, $category, $categoryPath, $pagedef)
     {
-        $queryParameter = $request->query->all();
-        if (0 === count($queryParameter)) {
-            $queryParameter = null;
-        }
-        $catPath = $category;
-        $catPrefix = $categoryPath;
-        $tmp = array();
-        if ('' !== $catPrefix) {
-            $tmp[] = $catPrefix;
-        }
-        if ('' !== $catPath) {
-            $tmp[] = $catPath;
-        }
-        $fullPath = '/'.implode('/', $tmp);
-
-        $aKey = array(
-            'class' => __CLASS__,
-            'method' => 'shopCategory',
-            'path' => $catPath,
-        );
-
-        $catPath = rtrim($catPath, '/');
+        $catPath = rtrim($category, '/');
         if ('.html' === substr($catPath, -5) && strlen($catPath) > 5) {
             $catPath = substr($catPath, 0, -5);
         }
 
-        $cache = $this->getCache();
-        $key = $cache->getKey($aKey);
-        $aResponse = $cache->get($key);
-        if (null !== $aResponse) {
-            return $this->processCategoryResponse($aResponse, $request);
-        }
-
-        $aResponse = array(
-            'activeShopArticle' => null,
-            'activeShopCategory' => null,
-            'redirectURL' => null,
-            'redirectPermanent' => false,
-            'noMatch' => false,
-            'pagedef' => $pagedef,
-            'queryParameter' => $queryParameter,
-        );
-
         $oPathCat = TdbShopCategoryList::GetCategoryForCategoryPath(explode('/', $catPath));
+
         if (null !== $oPathCat && true === $oPathCat->AllowDisplayInShop()) {
+            $aKey = array(
+                'class' => __CLASS__,
+                'method' => 'shopCategory',
+                'path' => $category,
+            );
+            $cache = $this->getCache();
+            $key = $cache->getKey($aKey);
+            $aResponse = $cache->get($key);
```

Comment:
I don't see groups here anywhere?

Reply:
That's the whole reason this PR exists. As the groups are not cached, we need to check accessibility independently of the cache. A different solution would have been to add the groups to the cache keys, but this could result in a cache that is overloaded with entries and make it virtually ineffective. So we decided to check for the accessibility instead.